core/node/consensus/src/storage/testonly.rs

```diff
@@ -189,3 +195,111 @@ impl ConnectionPool {
         Ok(())
     }
 }
+
+/// A struct for writing to consensus L2 contracts.
+#[derive(Debug)]
+pub struct VMWriter {
+    pool: ConnectionPool,
+    node: StateKeeper,
+    account: Account,
+}
+
+impl VMWriter {
+    /// Constructs a new `VMWriter` instance.
+    pub fn new(pool: ConnectionPool, node: StateKeeper, account: Account) -> Self {
+        Self {
+            pool,
+            node,
+            account,
+        }
+    }
+
+    /// Deploys the consensus registry contract and adds nodes to it.
+    pub async fn deploy_and_add_nodes(
+        &mut self,
+        ctx: &Ctx,
+        owner: Address,
+        nodes: &[&[Token]],
+    ) -> Address {
+        let registry_contract = consensus_l2_contracts::load_consensus_registry_contract_in_test();
+
+        let mut txs: Vec<Transaction> = vec![];
+        let deploy_tx = self.account.get_deploy_tx_with_factory_deps(
+            &registry_contract.bytecode,
+            Some(&[Token::Address(owner)]),
+            vec![],
+            TxType::L2,
+        );
+        txs.push(deploy_tx.tx);
+        for node in nodes {
+            let tx = self.gen_tx_add(&registry_contract.contract, deploy_tx.address, node);
+            txs.push(tx);
+        }
+        txs.push(
+            self.gen_tx_set_validator_committee(deploy_tx.address, &registry_contract.contract),
```

Comment:
Is there any rason why the `contract` and the `address` parameters are in opposite order?